crates/turbo-tasks-fs/src/watcher.rs

```diff
@@ -226,164 +216,127 @@ impl DiskWatcher {
             let mut event = rx.recv().map_err(|e| match e {
                 RecvError => TryRecvError::Disconnected,
             });
```

Comment:
I know this isn't part of your changes, but this `map_err` with a useless `match` was confusing for me to read.

```suggestion
            let mut event = rx.recv().or(Err(TryRecvError::Disconnected));
```